components/TxForm.tsx

```diff
@@ -37,94 +56,138 @@ export const TxForm = () => {
     ],
   });
   const { writeAsync: doErc20Approval } = useContractWrite(config);
-  const [recipient, setRecipient] = useState<string>(
-    '0zk1qyn0qa5rgk7z2l8wyncpynmydgj7ucrrcczhl8k27q2rw5ldvv2qrrv7j6fe3z53ll5j4fjs9j5cmq7mxsaulah7ykk6jwqna3nwvxudp5w6fwyg8cgwkwwv3g4'
-  );
+  const [recipient, setRecipient] = useState<string>(recipientAddress || '');
   const needsApproval =
-    tokenAddress !== ethAddress &&
+    selectedToken?.address !== ethAddress &&
     ethers.utils
-      .parseUnits(tokenAmount || '0', tokenDecimals)
-      .gt(tokenAllowances.get(tokenAddress || '') || BigNumber.from(0));
+      .parseUnits(tokenAmount || '0', selectedToken?.decimals)
+      .gt(tokenAllowances.get(selectedToken?.address || '') || BigNumber.from(0));
+
+  const onSubmit = handleSubmit(async (values) => {
+    setRecipient(values.recipient);
+    setTokenAmount(values.amount);
+    onReviewOpen();
+  });
+
+  const updateOnNetworkChange = useCallback(
+    (net: GetNetworkResult) => {
+      if (net && net?.chain) {
+        const chain = networks[net.chain.id];
+        const token = buildBaseToken(chain.baseToken, net.chain.id);
+        setSelectedToken(token);
```

Comment:
How come on network change we change the token, but nothing else? We could keep the token selection if the token also exists on the new chain. However if this is a nice to have, not a requirement, we can create an issue and track separately

Reply:
i'm a little confused here for different reasons. what does `updateNetworkChange` achieve?

is this complexity due to https://github.com/ScopeLift/railgun-shielder/pull/65#discussion_r1086971131 or something else?

Reply:
Yes, it is due to that comment. And this resets the the selected token to the native token on network change because it is the simplest way to guarantee a chain compatible token. We could add additional logic if we want to see if we can use the same token on the previous chain like Matt suggested, but I would say this is a nice to have and can be handled in a separate ticket.

Reply:
part of me thinks we should just let "no token" be selected after all, as the alternative seems a bit bloaty, and to what gain? will defer to you @alexkeating, what do you think?

Reply:
My vote is to keep it for now if the following is right. I think we will need some network switching logic regardless if it is filled out or not. 

1, If we have an empty value and a user selects a token and then they switch networks they could have an invalid token value unless we clear on network switch. And if we clear, we will need something like this to clear the form state on network change if that is the behavior we want instead.

Reply:
sounds good, but couldn't it be a simple `useEffect` hook where the callback sets the token state to `undefined`? or, is it more complicated now that we're using the ref with the react-hook-form? Anyway, I think i'll resolve this in favor of moving things along 👍